src/Lean/Server/Completion.lean

```diff
@@ -242,13 +242,23 @@ private def dotCompletion (ctx : ContextInfo) (info : TermInfo) (expectedType? :
             if (← isDotCompletionMethod typeName c) then
               addCompletionItem c.name.getString! c.type expectedType?
 
-private def optionCompletion (ctx : ContextInfo) : IO (Option CompletionList) := do
+private def optionCompletion (ctx : ContextInfo) (stx : Syntax) : IO (Option CompletionList) :=
   ctx.runMetaM {} do
-    let decls ← getOptionDecls
-    let opts ← getOptions
-    let items : Array CompletionItem := decls.fold (init := #[]) fun items name decl =>
-      items.push { label := name.toString, detail? := s!"({opts.get name decl.defValue}), {decl.descr}", documentation? := none }
-    return some { items := sortCompletionItems items, isIncomplete := true }
+      let partialName := stx[1].getId
+      -- HACK(WN): unfold the type so ForIn works
+      let (decls : Std.RBMap _ _ _) ← getOptionDecls
+      let opts ← getOptions
+      let mut items := #[]
+      for ⟨name, decl⟩ in decls do
+        if partialName.isPrefixOf name ||
+          (match partialName, name with
+            | Name.str p₁ s₁ _, Name.str p₂ s₂ _ => p₁ == p₂ && s₁.isPrefixOf s₂
+            | _, _ => false) then
+          items := items.push
+            { label := name.toString
+              detail? := s!"({opts.get name decl.defValue}), {decl.descr}"
```

Comment:
This is confusing, the default value may not be the current value. "(true), (pretty printer) ..." also looks a bit weird I think.

Reply:
Whoops, I thought I double checked that this was indeed new code :smile:

Reply:
I agree it looks a bit strange, but I also tried using the proper `documentation?` LSP field and with that you have to expand the popup to see the description, whereas with `details` it always shows. Note also that `opts.get name decl.defValue` is the *current* value (right?).

Reply:
Ah, you are right about the current value